File: src/main/java/com/cardlay/techinterview/cardmanagement/controller/request/CreateCardHolderDto.java
package com.cardlay.techinterview.cardmanagement.controller.request;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

public record CreateCardHolderDto(
    @NotNull
    @NotBlank
    String name,

    @NotNull
    @NotBlank
    @Email
    String email
) {
}

File: src/main/kotlin/com/cardlay/techinterview/cardmanagement/controller/request/CreateCardHolderDto.kt
package com.cardlay.techinterview.cardmanagement.controller.request

import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

data class CreateCardHolderDto(
    @field:NotNull
    @field:NotBlank
    val name: String,

    @field:NotNull
    @field:NotBlank
    @field:Email
    val email: String,
)